machine/tool/clang/src/main/java/org/qbicc/machine/tool/clang/ClangCCompilerInvokerImpl.java

```diff
@@ -82,7 +82,11 @@ void addArguments(final List<String> cmd) {
         Platform platform = getTool().getPlatform();
         cmd.add("-target");
         cmd.add(platform.getCpu().toString() + "-" + platform.getOs().toString() + "-" + platform.getAbi().toString());
-        Collections.addAll(cmd, "-std=gnu11", "-f" + "input-charset=UTF-8", "-pipe");
+        if (sourceLanguage == SourceLanguage.C) {
+            Collections.addAll(cmd, "-std=gnu11", "-f" + "input-charset=UTF-8");
+            cmd.add("-pthread");
```

Comment:
-pthread is definitely not used when compiling ASM files on MacOS.   I suspect this is true in general, but have not verified on other platforms.  It's not entirely clear from the documentation...

Reply:
It's weird that they ignore input charset for ASM files, but 🤷🏻‍♂️.